lib/declarative_authorization/authorization.rb

```diff
@@ -276,36 +278,73 @@ def permit!(privilege, options = {})
                   resource_type: "lease_document",
                   resource_id: options[:object]&.lease_document_uuid.to_s,
                   permission: "renewal",
-                  subject_type: "lease_document",
+                  subject_type: "lease_document"
                 )
-                actual_value = response[:subject_ids].any?
+                actual_value = response.any?
               end
 
               puts "  Is renewal - Rule requires: #{rule_requires}, Actual: #{actual_value}"
               if rule_requires == actual_value
-                condition_matched = true
                 puts "  ✓ Is_renewal condition matched!"
               else
                 puts "  ✗ Is_renewal condition did not match"
+                current_attribute_matched = false
+              end
+            end
+    
+            if conditions.key?(:id) && current_attribute_matched
+              condition_type = conditions[:id][0]
+              condition_proc = conditions[:id][1]
+    
+              if condition_type == :id_in_scope && condition_proc.is_a?(Proc)
+                puts "  Checking Occupancy ID in scope condition"
+                user_has_access_to_occupancy = attribute.validate?(attr_validator)
```

Comment:
When calling `attribute.validate` method, the `condition_proc` is executed, which returns a list of `Occupancy` `id`s the user has access to.  This logic exists in the `visible_by_conditions` method [here](https://github.com/appfolio/apm_bundle/pull/91595/files#diff-f1b5946b88d756af239aadf350499b40e554412306abcd6000c0bb94f4de68beR10)